instrumentation/opentelemetry-api/opentelemetry-api-1.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/ContextBridgeTest.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.opentelemetryapi;
+
+import static io.opentelemetry.api.common.AttributeKey.stringKey;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.annotations.WithSpan;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.semconv.incubating.CodeIncubatingAttributes;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicReference;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@ExtendWith(AgentInstrumentationExtension.class)
+public class ContextBridgeTest {
+
+  @RegisterExtension
+  private static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static final ContextKey<String> ANIMAL = ContextKey.named("animal");
+
+  @Test
+  @DisplayName("agent propagates application's context")
+  public void agentPropagatesApplicationsContext() throws Exception {
+    // When
+    Context context = Context.current().with(ANIMAL, "cat");
+    AtomicReference<String> captured = new AtomicReference<>();
+    try (Scope ignored = context.makeCurrent()) {
+      Executors.newSingleThreadExecutor()
+          .submit(() -> captured.set(Context.current().get(ANIMAL)))
+          .get();
+    }
+
+    // Then
+    assertEquals("cat", captured.get());
```

Comment:
we typically will use `org.assertj.core.api.Assertions.assertThat` for assertions.

for example:
```
assertThat(captured.get()).isEqualTo("cat");
```